torchquantum/pulse/quantum_pulse_simulation.py

```diff
@@ -0,0 +1,183 @@
+import torch
+import torch.optim as optim
+import torchquantum as tq
+import torchquantum.functional as tqf
+import numpy as np
+
+class QuantumPulseDemo(tq.QuantumModule):
+    """
+    Quantum pulse demonstration module.
+
+    This module defines a parameterized quantum pulse and applies it to a quantum device.
+    """
+
+    def __init__(self):
+        """
+        Initializes the QuantumPulseDemo module.
+        """
+        super().__init__()
+        self.n_wires = 2
```

Comment:
I would recommend parametrising the function. If you're doing an example, it should be in the examples folder. I'll wait for @Hanrui-Wang to review for his suggestions.